services/frontend/src/components/PersonalDetailsDialog/PersonalDetailsDialog.js

```diff
@@ -1,48 +1,69 @@
-import React from "react";
+import React, {useCallback} from "react";
 import {useInput} from "utils/hooks";
 import {Button, Dialog, DialogActions, DialogContent, DialogTitle} from "@material-ui/core";
 import {PersonalDetailsForm} from "components/";
 import "./PersonalDetailsDialog.scss";
 
-const PersonalDetailsDialog = ({profile, handleClose, open}) => {
+const PersonalDetailsDialog = ({profile, closeDialog, open, onSubmit, error}) => {
+    /* setup controlled fields with react hooks */
     const formFieldData = {
-        "nameInput": {
-            ...useInput(profile.name)
+        "name": {
+            ...useInput(profile.name),
         },
-        "emailInput": {
+        "contactEmail": {
             ...useInput(profile.contactEmail)
         },
-        "roleInput": {
+        "primaryRole": {
             ...useInput(profile.primaryRole)
         },
-        "phoneInput": {
+        "phone": {
             ...useInput(profile.phone)
         },
-        "slackInput": {
+        "slackHandle": {
             ...useInput(profile.slackHandle)
         },
-        "rocketChatInput": {
+        "rocketChatHandle": {
             ...useInput(profile.rocketChatHandle)
         }
     };
 
```

Comment:
When you change a field and then click cancel. Then reopen it, the changed value is there not the original value. I think that the field should be reset when you hit cancel.

Reply:
Well it's a really hacky but I changed it to work like that.

changed 
```jsx
    key={profile.id}
```
to 
```jsx
    key={personalDetailsDialogOpen ? profile : error}
```
in ProfileLayout.js ln116.